src/rules/strictComparisonsRule.ts

```diff
@@ -120,9 +120,68 @@ export class Rule extends Lint.Rules.TypedRule {
     }
 }
 
+function getConstraint(t: ts.Type): ts.TypeReferenceNode | undefined {
+    if (t.getSymbol() === undefined) {
+        return undefined;
+    }
+
+    const paramDeclaration = t.symbol.declarations.find(
+        d => d.kind === ts.SyntaxKind.TypeParameter,
+    ) as ts.TypeParameterDeclaration;
+    if (paramDeclaration === undefined) {
+        return undefined;
+    }
+
+    const constraint = getEffectiveConstraintOfTypeParameter(paramDeclaration);
+    if (constraint === undefined) {
+        return undefined;
+    }
+
+    if (ts.isTypeReferenceNode(constraint)) {
+        return constraint;
+    }
+
+    return undefined;
+}
+
+function findValueDeclarationOfType(typeName: string, sourceFile: ts.SourceFile) {
+    const locals = (sourceFile as any).locals as Map<string, ts.Symbol>;
+    const type = locals.get(typeName);
+    if (type !== undefined) {
+        return type.valueDeclaration;
+    }
+    return undefined;
+}
+
+function findIdentifiers(expressions: ts.Expression[]): ts.Identifier[] {
+    return expressions
+        .map(e => {
+            if (isBinaryExpression(e)) {
+                return findIdentifiers([e.left, e.right]);
+            }
+            return e;
+        })
+        .reduce<ts.Identifier[]>((acc, val) => acc.concat(val as ts.Identifier[]), []);
```

Comment:
A `flatMap` utility exists in `src/utils.ts`, maybe that would simplify this a bit?